app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -41,38 +45,99 @@ export default class SchoolDisciplineDashboard extends React.Component {
-    let studentDisciplineIncidentCounts = {};
-
-    //if a user selects a category and then moves to a time range with no incidents within that category, return an empty object
-    if (!incidents) return studentDisciplineIncidentCounts;
-    incidents.forEach((incident) => {
-      studentDisciplineIncidentCounts[incident.student_id] = studentDisciplineIncidentCounts[incident.student_id] || 0;
-      studentDisciplineIncidentCounts[incident.student_id]++;
+  filterStudents(students) {
+    const {selectedChart, selectedCategory} = this.state;
+    return students.filter(student => {
+      if (selectedCategory && selectedChart === 'grade' && student.grade !== selectedCategory) return false;
+      if (selectedCategory && selectedChart === 'race' && student.race !== selectedCategory) return false;
+      if (selectedCategory && selectedChart === 'homeroom_label' && student.homeroom_label !== selectedCategory) return false;
+      if (selectedCategory && !this.filterIncidents(student.discipline_incidents, true).length) return false;
+      return true;
     });
-    return studentDisciplineIncidentCounts;
   }
 
-  sortChartKeys(groupedIncidents) {
-    const chartKeys = Object.keys(groupedIncidents);
-    switch(this.state.selectedChart) {
-    case 'time': return this.sortedTimes(chartKeys);
-    case 'day': return this.sortedDays(chartKeys);
-    case 'grade': return this.sortedGrades(chartKeys);
-    default: return this.sortedByIncidents(groupedIncidents); //because number of incidents in each category needed here
+  timeStampToHour(incident) {
+    return incident.has_exact_time ? moment.utc(incident.occurred_at).startOf('hour').format('h:mm a') : "Not Logged";
+  }
+
+  timeStampToDay(incident) {
+    return moment.utc(incident.occurred_at).format("ddd");
+  }
+
+  getIncidentsFromStudents(students) {
+    //for chart data, does not filter on selected categories
+    return _.flatten(students.map(student => this.filterIncidents(student.discipline_incidents, false)));
+  }
+
+  //Depending on the chart, incidents are grouped either by student attribute or incident attribute. Any future
+  //charts may be handled here.
+  getChartData(students, group) {
+    switch(group) {
+    case 'homeroom_label': case 'race': {
+      const groupedStudents = _.groupBy(students, group);
+      const categories = this.sortedByStudent(groupedStudents);
+      const seriesData = categories.map(category => {
+        return this.getIncidentsFromStudents(groupedStudents[category]).length;
+      });
+      return {categories, seriesData};
+    }
+    case 'grade': {
+      const groupedStudents = _.groupBy(students, group);
+      const categories = this.sortedGrades(Object.keys(groupedStudents));
+      const seriesData = categories.map(category => {
+        return this.getIncidentsFromStudents(groupedStudents[category]).length;
+      });
+      return {categories, seriesData};
+    }
+    case 'incident_code': case 'incident_location': {
```

Comment:
Should `incident_code` be removed, since it's now a filter up top instead?

Reply:
I meant to ask about that. I wasn't sure we necessarily wanted to remove the ability to compare different incident types, though I'm not sure there's much useful information to be gained from that.

Reply:
Yeah, if it's not being used let's remove it.  We've got source control so always easy to bring things back :)